src/contrib/mlir/pass/mlir_subgraph_extraction.cpp

```diff
@@ -108,6 +270,30 @@ bool MLIRSubgraphExtractionPass::is_supported_mlir_op(std::shared_ptr<Node> node
     return true;
 }
 
+bool MLIRSubgraphExtractionPass::check_cycles(NodeVector& inputs, std::vector<int>& subgraph_ids)
+{
+    NodeVector work_list;
+    NGRAPH_DEBUG << "[CK Extract] Inputs size: " << inputs.size() << std::endl;
+    work_list.insert(work_list.end(), inputs.begin(), inputs.end());
+    while (!work_list.empty())
+    {
+        auto node = work_list.back();
+        work_list.pop_back();
+        if (std::find(subgraph_ids.begin(), subgraph_ids.end(), get_subgraph(node)) !=
+            subgraph_ids.end())
+        {
+            // we hit one of the sub-graphs we want to extend. we have a cycle.
+            NGRAPH_DEBUG << "[CK Extract] Cycle found when trying to add node" << std::endl;
+            return true;
```

Comment:
I think we shouldn't return here. This is just telling us that we found a cycle in one of the potential subgraphs we want to extend but we should try to find another subgraph we can extend instead of bailing out after the first cycle found, right?

Reply:
we aren't bailing out, we are just saying a cycle is found. If you check the call-site we actually start a new sub-graph.

Reply:
Yeah, that's what I meant by "bailing out": we are creating a new graph instead of trying to find an existing one in the remaining sub-graphs where the op doesn't create a cycle. It could be a future improvement but, please could you add a comment about this?